include/shelf-pack.hpp

```diff
@@ -303,19 +474,69 @@ class ShelfPack {
     int32_t width() const { return width_; }
     int32_t height() const { return height_; }
 
+
 private:
-    void count(int32_t h) {
-       stats_[h]++;
+
+    /**
+     * Called by packOne() to allocate a bin by reusing an existing freebin
+     *
+     * @private
+     * @param    {Bin&}       bin    Reference to a freebin to reuse
+     * @param    {int32_t}    w      Width of the bin to allocate
+     * @param    {int32_t}    h      Height of the bin to allocate
+     * @param    {int32_t}    id     Unique identifier for this bin
+     * @returns  {Bin*}       Pointer to a Bin with `id`, `x`, `y`, `w`, `h` properties
+     *
+     * @example
+     * Bin* bin = sprite.allocFreebin(freebin, 12, 16, 5);
+     */
+    Bin* allocFreebin(Bin& bin, int32_t id, int32_t w, int32_t h) {
+        freebins_.erase(std::remove(freebins_.begin(), freebins_.end(), &bin), freebins_.end());
```

Comment:
> You could replace the bin parameter with a freebins_ iterator, and then do freebins_.erase(it)

Are vector iterators stable enough?  I'm thinking if another thread `unrefs` a bin and `freebins` is changed.  (is tile loading/unloading multithreaded?  This is the usecase)


> or even better, a O(1) swap-with-last-element-and-truncate operation.

Confused by this.. Because they are different sizes, we're searching for a _specific_ bin to reuse, not just taking the last element.